packages/controller/lib/setup/setupBackup.js

```diff
@@ -670,6 +679,52 @@ class BackupRestore {
             return EXIT_CODES.CANNOT_RESTORE_BACKUP;
         }
 
+        const controllerDir = tools.getControllerDir();
+
+        // check that the same controller version is installed as it is contained in backup
+        if (!force) {
+            const backupHostname = restore.config.system.hostname || hostname;
+
+            try {
+                const ioPackJson = fs.readJsonSync(path.join(controllerDir, 'io-package.json'));
+                const backupJson = JSON.parse(data);
+                const hostObj = backupJson.objects.find(obj => obj.id === `system.host.${backupHostname}`);
+                if (hostObj.value.common.installedVersion !== ioPackJson.common.version) {
+                    console.warn('The current version of js-controller differs from the version in the backup.');
+                    console.warn('The js-controller version of the backup can not be restored automatically.');
+                    console.warn(
+                        `To restore the js-controller version of the backup, execute npm i iobroker.js-controller@${hostObj.value.common.installedVersion} --production inside your ioBroker directory`
+                    );
+                    console.warn(
+                        'If you really want to restore the backup with the current installed js-controller, execute the restore command with the --force flag'
+                    );
+
+                    return EXIT_CODES.CANNOT_RESTORE_BACKUP;
+                }
+            } catch {
+                // ignore
+            }
+        }
+
+        const nodeModulePath = path.join(controllerDir, '..');
+        const nodeModuleDirs = fs.readdirSync(nodeModulePath);
+
+        // we need to uninstall current adapters to get exact the same system as before backup
+        for (const dirName of nodeModuleDirs) {
+            if (
+                dirName.startsWith(`${tools.appName.toLowerCase()}.`) &&
+                dirName !== `${tools.appName.toLowerCase()}.js-controller`
+            ) {
+                try {
+                    const packJson = fs.readJsonSync(path.join(nodeModulePath, dirName, 'package.json'));
+                    console.log(`Removing current installation of ${packJson.name}`);
+                    await tools.uninstallNodeModule(packJson.name);
+                } catch {
+                    // ignore
```

Comment:
I would log a line "could not delete ... bla ... please executa "iob upload name"

Reply:
all adapters will be uploaded later anyways

Reply:
https://github.com/ioBroker/ioBroker.js-controller/blob/506b23dd40a5402f57846faa3e79b07659d09398/packages/controller/lib/setup/setupBackup.js#L596

Reply:
ahh ok ... hm ... but why this was then not happening before? In fact the missing eta data ala upload is one of the reasons for this topic